libraries/AP_Beacon/AP_Beacon_Marvelmind.cpp

```diff
@@ -0,0 +1,399 @@
+        return;
+
+    hedge->positionsBeacons.numBeacons = (n_used + 1);
+    b = hedge->positionsBeacons.beacons[n_used];
+}
+
+void AP_Beacon_Marvelmind::process_beacons_positions_datagram(
+        struct StationaryBeaconPosition &b) {
+    uint8_t n = input_buffer[5]; // number of beacons in packet
+    uint8_t i, ofs;
+    uint8_t address;
+    int16_t x, y, z;
+
+    if ((1 + n * 8) != input_buffer[4])
+        return; // incorrect size
+
+    for (i = 0; i < n; i++) {
+        ofs = 6 + i * 8;
+
+        address = input_buffer[ofs + 0];
+        x = input_buffer[ofs + 1] | (((uint16_t) input_buffer[ofs + 2]) << 8);
+        y = input_buffer[ofs + 3] | (((uint16_t) input_buffer[ofs + 4]) << 8);
+        z = input_buffer[ofs + 5] | (((uint16_t) input_buffer[ofs + 6]) << 8);
+
+        getOrAllocBeacon(b, address);
+        if (b.address != NULL) {
+            b.address = address;
+            b.x = x * 10; // millimeters
+            b.y = y * 10; // millimeters
+            b.z = z * 10; // millimeters
+
+            b.highResolution = false;
+
+            hedge->positionsBeacons.updated = true;
+        }
+    }
+}
+
+void AP_Beacon_Marvelmind::process_beacons_positions_highres_datagram(
+        struct StationaryBeaconPosition &b) {
+    uint8_t n = input_buffer[5]; // number of beacons in packet
+    uint8_t i, ofs;
+    uint8_t address;
+    int32_t x, y, z;
+
+    if ((1 + n * 14) != input_buffer[4])
+        return; // incorrect size
+
+    for (i = 0; i < n; i++) {
+        ofs = 6 + i * 14;
+
+        address = input_buffer[ofs + 0];
+        x = input_buffer[ofs + 1] | (((uint32_t) input_buffer[ofs + 2]) << 8)
+                | (((uint32_t) input_buffer[ofs + 3]) << 16)
+                | (((uint32_t) input_buffer[ofs + 4]) << 24);
+        y = input_buffer[ofs + 5] | (((uint32_t) input_buffer[ofs + 6]) << 8)
+                | (((uint32_t) input_buffer[ofs + 7]) << 16)
+                | (((uint32_t) input_buffer[ofs + 8]) << 24);
+        z = input_buffer[ofs + 9] | (((uint32_t) input_buffer[ofs + 10]) << 8)
+                | (((uint32_t) input_buffer[ofs + 11]) << 16)
+                | (((uint32_t) input_buffer[ofs + 12]) << 24);
+
+        getOrAllocBeacon(b, address);
+        if (b.address != NULL) {
+            b.address = address;
+            b.x = x;
+            b.y = y;
+            b.z = z;
+
+            b.highResolution = true;
+
+            hedge->positionsBeacons.updated = true;
+        }
+    }
+}
+
+uint8_t AP_Beacon_Marvelmind::markPositionReady() {
+    uint8_t ind = hedge->lastValues_next;
+    uint8_t indCur = ind;
+
+    hedge->positionBuffer[ind].ready = true;
+    hedge->positionBuffer[ind].processed = false;
+    ind++;
+    if (ind >= hedge->maxBufferedPositions)
+        ind = 0;
+    if (hedge->lastValuesCount_ < hedge->maxBufferedPositions)
+        hedge->lastValuesCount_++;
+    hedge->haveNewValues_ = true;
+
+    hedge->lastValues_next = ind;
+
+    return indCur;
```

Comment:
indCur is return without have been used ? Is something missing ?
if not, make it const !